exist-core/src/main/java/org/exist/client/NewResourceDialog.java

```diff
@@ -287,7 +287,7 @@ private void createResource(final ResourceType resourceType, final String filena
         
         try {
             final String resName = URIUtils.urlEncodeUtf8((isNullOrEmpty(filename) ? DEFAULT_FILENAME : filename) + "." + resourceType.getFileExtension());
-            final String resType = resourceType == ResourceType.XML_DOCUMENT ? XMLResource.RESOURCE_TYPE : BinaryResource.RESOURCE_TYPE;
+            final Class<? extends Resource> resType = resourceType == ResourceType.XML_DOCUMENT ? XMLResource.class : BinaryResource.class;
```

Comment:
for readability split over lines?

Reply:
I don't seem to get your point here... The `resType` is now of type `Class` type not a `String` any more and the generic definition is needed in order to work for the later call.

Reply:
I believe @dizzzz was only referring to the formatting.
Maybe he had something like this in mind.

```
final Class<? extends Resource> resType = resourceType == ResourceType.XML_DOCUMENT 
    ? XMLResource.class
    : BinaryResource.class;
```

Reply:
I see... check for that with the next batch of changes then